core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
-        } else {
+        } else if (fir.isParam) {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
+        } else {
+          Nil
```

Comment:
Characterize the difference in Foo and Bar for me in that example? Is it that the reference in Bar.toString causes the scala compiler to generate a field for x0 and in Foo it  doesn't generate a field?

The code looks to me like currently for a field it always tries to generate a putField, but with your change it only tries to generate getField if the field `isParam` (which means it's a constructor parameter it looks like). But that doesn't seem to match what you're saying about `x0` in the example? what am I missing?